### PR TITLE
Add Spark MAX Factory

### PR DESCRIPTION
**Pull request by StephenWelch**
Only configures settings with obvious defaults until we can do a parameter dump

### CODE REVIEW

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*----------------------------------------------------------------------------*/
```
Comment (flybotix) [inline robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java]: is ILITE code really copyrighted by FIRST?

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * 
+ */
+public class SparkMaxFactory {
+
+    public static class Configuration {
+        public int CAN_TIMEOUT = 100;
+        public int CONTROL_FRAME_PERIOD = 5;
+        public IdleMode IDLE_MODE = IdleMode.kBrake;
+        public boolean IS_INVERTED = false;
+        public int STATUS_0_PERIOD_MS = 10;
```
Comment (flybotix) [inline robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java:26]: Make sure these are the default frame periods; we don't want to auto-overload the CAN bus

**Review by flybotix — CHANGES_REQUESTED** [review]

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java (hunk 1 above)
Comment (StephenWelch) [reply robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java]: The header is automatically added when you create a new class in VSCode, would you like me to remove it?

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java (hunk 2 above)
Comment (StephenWelch) [reply robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java:26]: I set these to the same values as the TalonSRXFactory, would you like me to change it to 10? (That's the default) My train of thought was that the cost of setting the master controller to a higher rate would be offset by setting the slave control frame period to 100ms.

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java (hunk 1 above)
Comment (flybotix) [reply robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java]: Yes please

Comment (flybotix) [reply robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java]: (not a blocker to this review)

File: robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java (hunk 2 above)
Comment (flybotix) [reply robot/src/main/java/us/ilite/lib/drivers/SparkMaxFactory.java:26]: I'd set it to the default from Rev, then we can adjust as needed.  If we haven't tested full utilization, there isn't any reason to change from the defaults (except for specific cases of sensors).

**Review by flybotix — APPROVED** [review]

**Pull request merged**